### PR TITLE
Add implementation text for reactors.py.

### PR DESCRIPTION
**Pull request by mgjarrett**
## What is the change?

Design and implementation detail added for reactors.py

Also updated some docstrings in a related function in a different file.

## Why is the change being made?

This is part of ongoing SQA effort.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/reactors.py [hunk 1]
```diff
@@ -348,6 +379,18 @@ def symmetry(self) -> geometry.SymmetryType:
...
+            This property getter returns the symmetry attribute of the
+            spatialGrid instance attribute.  The spatialGrid is an instance of a
+            child of the abstract base class :py:class:`Grid
+            <armi.reactor.grids.grid.Grid>` type.  The symmetry attribute is an
+            instance of the :py:class:`SymmetryType
+            <armi.reactor.geometry.SymmetryType>` class, which is a wrapper
+            around the :py:class:`DomainType <armi.reactor.geometry.DomainType>`
+            and :py:class:`BoundaryType <armi.reactor.geometry.BoundaryType>`
+            enumerations used to classify the domain (e.g., 1/3 core, quarter
+            core, full core) and symmetry boundary conditions (e.g., periodic,
```
Comment (jyang-TP) [inline armi/reactor/reactors.py:392]: 1/3 core works only with periodic boundary condition. Quarter core works with either periodic or reflective boundary condition.  We may want to make it clear here.

File: armi/reactor/reactors.py [hunk 2]
```diff
@@ -348,6 +379,18 @@ def symmetry(self) -> geometry.SymmetryType:
...
+            reflective, none) of a reactor, respectively.
```
Comment (jyang-TP) [inline armi/reactor/reactors.py:393]: 1/3 core works only with periodic boundary condition. Quarter core works with either periodic or reflective boundary condition. We may want to make it clear here.

File: armi/reactor/reactors.py [hunk 3]
```diff
@@ -1651,6 +1707,18 @@ def getAssemblyWithStringLocation(self, locationString):
...
+            location. The location is provided to this method as an input
+            parameter in a string with the format "001-001". For a `HexGrid
+            <armi.reactor.grids.hexagonal.HexGrid>`, the first number indicates
+            the hexagonal ring and the second number indicates the position
+            within that ring. For a `CartesianGrid
+            <armi.reactor.grids.cartesian.CartesianGrid>`, the first number
+            represents the x index and the second number represents the y index.
+            If there is no assebly in the grid at the requested location, this
+            method returns None.
         """
```
Comment (jyang-TP) [inline armi/reactor/reactors.py:1722]: A typo:  If there is no assembly .....

File: armi/reactor/reactors.py [hunk 4]
```diff
@@ -1679,68 +1747,102 @@ def findNeighbors(
...
+            The `showBlanks` option determines whether non-existing assemblies
+            will be indicated with a `None` in the list or just excluded from
+            the list altogether.
+
+            The `duplicateAssembliesOnReflectiveBoundary` setting only works for
+            1/3 core symmetry with periodic boundary conditions. For these types
+            of geometries, if this setting is `True`, neighbor lists for
+            assemblies along a periodic boundary will include the assemblies
+            along the opposite periodic boundary that are effectively neighbors.
+
```
Comment (jyang-TP) [inline armi/reactor/reactors.py:1781]: Did we use 1/3 core symmetry with Cartesian grid?  No test for it.

**Comment by john-science** [comment]
Sorry to touch your PR.  I just merged in the `main` branch, because there was a conflict.

File: armi/reactor/reactors.py [hunk 5]
```diff
@@ -1679,68 +1747,102 @@ def findNeighbors(
...
+        For a Cartesian grid, the order of the neighbors is east, north, west,
+        south.
 
         .. impl:: Retrieve neighboring assemblies of a given assembly.
             :id: I_ARMI_R_FIND_NEIGHBORS
             :implements: R_ARMI_R_FIND_NEIGHBORS
 
+            This method takes an :py:class:`Assembly
+            <armi.reactor.assemblies.Assembly` as an input parameter and returns
```
Comment (john-science) [inline armi/reactor/reactors.py:1763]:
```suggestion
            <armi.reactor.assemblies.Assembly>` as an input parameter and returns
```

File: armi/reactor/reactors.py (hunk 2 above)
Comment (mgjarrett) [reply armi/reactor/reactors.py:393]: I added a paragraph about valid geometry and symmetry combinations

File: armi/reactor/reactors.py (hunk 4 above)
Comment (mgjarrett) [reply armi/reactor/reactors.py:1789]: There is an input validator that checks the combinations of symmetry and geometry type, which will prevent a user from requesting 1/3 core symmetry with Cartesian grid.

https://github.com/terrapower/armi/blob/14c951a05214154a1e6bfb43c728aa714084cea7/armi/operators/settingsValidation.py#L722-L733

**Review by jyang-TP — APPROVED** [review]

**Pull request merged**